resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -175,14 +175,22 @@ trait MesosSchedulerUtils extends Logging {
     registerLatch.countDown()
   }
 
-  def createResource(name: String, amount: Double, role: Option[String] = None): Resource = {
+  def createResource(
+       name: String,
+       amount: Double,
+       role: Option[String] = None,
+       allocationInfo: Option[AllocationInfo] = None,
+       reservationInfo: Option[ReservationInfo] = None): Resource = {
     val builder = Resource.newBuilder()
       .setName(name)
       .setType(Value.Type.SCALAR)
       .setScalar(Value.Scalar.newBuilder().setValue(amount).build())
 
     role.foreach { r => builder.setRole(r) }
-
+    if (role.forall(r => !r.equals("*"))) {
```

Comment:
Make a constant for the star role (`*`) called `ANY_ROLE`

Reply:
ok sounds good.